src/pipelines/html.rs

```diff
@@ -57,8 +57,10 @@ impl HtmlPipeline {
     }
 
     /// Spawn a new pipeline.
-    pub fn spawn(self: Arc<Self>) -> JoinHandle<Result<()>> {
-        spawn_local(self.build())
+    pub async fn spawn(self: Arc<Self>) -> Result<()> {
+        // let local = &self.clone().local;
+        // local.spawn_local(self.build())
+        self.build().await
```

Comment:
I initially used `spawn_local` but that panicked because there was no local task set. I might be able to use `spawn`, not sure about that. Should that be used?

Reply:
Actually, I just tested using `spawn` there. It doesn't work because the future returned by `self.build()` is not `Send`